phylogenetic/rules/merge_sequences.smk

```diff
@@ -0,0 +1,189 @@
+"""
+This part of the workflow merges inputs from any of:
+    curated NCBI dataset: via s3 remote
+    additional local data: via path to files
+
+based on what is defined in the config YAML file.
+
+REQUIRED INPUTS:
+
+    config.yaml  = defines input files within a dictionary in either 'inputs' or 'additional_inputs'
+
+OUTPUTS:
+
+    input_sequences = gathered and merged sequences
+    input_metadata  = gathered and merged metadata
+
+
+This part of the workflow usually includes the following steps:
+
+    - Any transformation to match the columns of the tsv files
+    - Concatenation of the tsv and the sequences files
+
+"""
+
+
+# ------------- helper functions to collect, merge & download input files ------------------- #
+NEXTSTRAIN_PUBLIC_BUCKET = "s3://nextstrain-data/"
+
+def _parse_config_input(input):
+    """
+    Parses information from an individual config-defined input, i.e. an element within `config.inputs` or `config.additional_inputs`
+    and returns information snakemake rules can use to obtain the underlying data.
+
+    The structure of `input` is a dictionary with keys:
+    - name:string (required)
+    - metadata:string (optional) - a s3 URI or a local file path
+    - sequences:string (optional) - a s3 URI or a local file path
+
+    Returns a dictionary with optional keys:
+    - metadata:string - the relative path to the metadata file. If the original data was remote then this represents
+      the output of a rule which downloads the file
+    - metadata_location:string - the URI for the remote file if applicable else `None`
+    - sequences:string - the relative path to the sequences FASTA. If the original data was remote then this represents
+      the output of a rule which downloads the file
+    - sequences_location:string - the URI for the remote file if applicable else `None`
+
+    Raises InvalidConfigError
+    """
+    name = input['name']
+    lambda_none = lambda w: None
+
+    info = {'metadata': None, 'metadata_location': None, 'sequences': lambda_none, 'sequences_location': lambda_none}
```

Comment:
The `lambda_none` can go - it's a holdover from the segment-specific complexity of sequence inputs. With the current code, using an input without sequences, e.g. USVI with only metadata, raises an error:

```
WorkflowError in rule merge_sequences in file /Users/naboo/github/nextstrain/zika/phylogenetic/rules/merge_sequences.smk, line 172:
Function did not return str or list of str.
```